### PR TITLE
Added python check and auto fix scripts

### PR DESCRIPTION
**Pull request by HU90m**
We now have a python lints and an auto-formatter, with the lints running on pull requests.

The python auto-formatter is set up to comply with the lowRISC/OpenTitan style guide (PEP8).

```
nix run .#python_check
nix run .#python_fix
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,16 +10,42 @@ In the meantime, please read the [architecture specfication](./doc/architecture/
 
 Symphony is part of the [Sunburst Project](https://www.sunburst-project.org) funded by [UKRI](https://www.ukri.org/) / [DSbD](https://www.dsbd.tech/).
 
-## Building Documentation
+## Contributing
+### Installing Nix
```
[the remark this answers is not in the document]
Comment (GregAC) [reply README.md:14]: Please just make sure that whatever way it's explained here it's as simple as possible. We'll have lots of people looking at this who will have no idea what nix is nor really care, but will want to be able to run the builds and lint so they can put a PR together. So there's no need to dive into minutiae here (such as why we need experimental features, that can live elsewhere in the docs) and ideally have a way to setup nix with the required experimental features with a single command.

**Review by GregAC — APPROVED** [review]

**Review by marnovandermaas — APPROVED** [review]
This looks reasonable to me. Please make sure everything is always still runnable without Nix.

**Comment by hcallahan-lowrisc** [comment]
@HU90m I don't think the comments were addressed before this one got merged. Are you planning to do so in a follow-up PR?

**Comment by HU90m** [comment]
> @HU90m I don't think the comments were addressed before this one got merged. Are you planning to do so in a follow-up PR?

Yep, sorry, the suggested doc changes are in: https://github.com/lowRISC/symphony-system/pull/6

**Pull request merged**